src/TemplateManager.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace RebelCode\Bookings\Wordpress\Module;
+
+use Psr\Container\ContainerInterface;
+use Psr\EventManager\EventManagerInterface;
+
+/**
+ * Class TemplateManager
+ *
+ * Template manager is responsible for registering templates in given
+ * event manager from container.
+ *
+ * It register all defined templates as actions in event manager. All templates
+ * are prefixed with `eddbk` or any other given prefix. We need prefix when we
+ * tweaking from outside.
+ *
+ * In this module we can work without referencing to the
+ * prefix by using `render($templateActionName)` method.
+ *
+ * Inside render method we will apply prefix and trigger action in event manager after this.
+ *
+ * @package RebelCode\Bookings\Wordpress\Module
+ */
+class TemplateManager
+{
+    /**
+     * @var EventManagerInterface
+     */
+    protected $eventManager;
+
+    /**
+     * Template prefix
+     *
+     * @var
+     */
+    protected $prefix;
+
+    /**
+     * TemplateManager constructor.
+     *
+     * @param ContainerInterface $c
+     * @param string $prefix
+     * @throws \Psr\Container\ContainerExceptionInterface
+     * @throws \Psr\Container\NotFoundExceptionInterface
+     */
+    public function __construct(ContainerInterface $c, $prefix = 'eddbk')
+    {
+        $this->eventManager = $c->get('event-manager');

```

Comment:
Since the container is not saved in this class, consider accepting an `EventManagerInterface` as the first parameter rather than a `ContainerInterface`. The `TemplateManager` can then be instantiated as:

```
new TemplateManager($c->get('event_manager'), $prefix);
```